src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1413,7 +1417,114 @@ private IBlockOperation CreateBoundBlockOperation(BoundBlock boundBlock)
             ITypeSymbol type = null;
             Optional<object> constantValue = default(Optional<object>);
             bool isImplicit = boundBlock.WasCompilerGenerated;
-            return new CSharpLazyBlockOperation(this, boundBlock, locals, _semanticModel, syntax, type, constantValue, isImplicit);
+            var nullChecksToPrepend = ImmutableArray<ConditionalOperation>.Empty;
+            bool blockInModel = _semanticModel.Root.ChildNodes().Contains(syntax) || _semanticModel.Root.Parent?.ChildNodes().Contains(syntax) == true;
+
+            if (_semanticModel is MemberSemanticModel memberModel
+                && memberModel.ContainsNullCheckedParameter
```

Comment:
>&& memberModel.ContainsNullCheckedParameter [](start = 16, length = 43)

This check seems unnecessary. Any reason we shouldn't call `GenerateNullChecksForParameters()` always?